module-new/src/components/switch/switch.component.tsx

```diff
@@ -0,0 +1,143 @@
+import { Root, SwitchProps, Thumb } from '@radix-ui/react-switch';
+import * as React from 'react';
+
+import { IBindingProps, useBindingState, ValidationMessage } from '../../hooks/form';
+import { ArmstrongFCExtensions, ArmstrongFCProps, ArmstrongFCReturn, NullOrUndefined } from '../../types';
+import { concat } from '../../utils/classNames';
+import { useArmstrongConfig } from '../config/config.context';
+import { IInputWrapperProps } from '../inputWrapper/inputWrapper.component';
+import { Label } from '../label/label.component';
+import { ValidationErrors } from '../validationErrors/validationErrors.component';
+
+import './switch.theme.css';
+
+export interface ISwitchProps<TBind extends NullOrUndefined<boolean>>
+  extends Omit<
+      React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>,
+      'onChange' | 'checked'
+    >,
+    Pick<
+      IInputWrapperProps,
+      | 'scrollValidationErrorsIntoView'
+      | 'validationMode'
+      | 'validationErrorMessages'
+      | 'error'
+      | 'pending'
+      | 'disabled'
+      | 'className'
+    > {
+  /**  prop for binding to an Armstrong form binder (see forms documentation) */
+  bind?: IBindingProps<TBind>;
+
+  /** (Optional) Whether the switch is checked or not */
+  checked?: TBind;
+
+  /** (Optional) A callback function (newValue: TData) => void to handle state when 'checked' is changed. */
+  onCheckedChange?: (newValue: TBind) => void;
+
+  /** (Optional) called when the user updates the switch value */
+  onChange?: (newValue: TBind) => void;
+
+  /** (Optional) The state of the switch when it is initially rendered. Use when you do not need to control its state. */
+  defaultChecked?: boolean;
+
+  /** (Optional) A boolean flag to disable the checkbox input. */
+  disabled?: boolean;
+
+  /** (Optional) A React.ReactNode to display a label for the switch input. */
+  label?: React.ReactNode;
+
+  /** (Optional) Classname for the switch label. */
+  labelClassName?: string;
+
+  /** (Optional) Indicates if switch is required */
+  required?: boolean;
+
+  /** (Optional) Can be a string or {key, element} key is necessary for animating in new messages   */
+  validationErrorMessages?: ValidationMessage[];
+
+  /** (Optional) A boolean flag to automatically scroll validation error messages into view. */
+  scrollValidationErrorsIntoView?: boolean;
+
+  /** (Optional) Classname for the validation errors */
+  validationErrorsClassName?: string;
+}
+
+export const Switch = React.forwardRef<HTMLButtonElement, ISwitchProps<NullOrUndefined<boolean>>>(
+  (
+    {
+      bind,
+      checked,
+      onCheckedChange,
+      defaultChecked,
+      disabled,
+      labelClassName,
+      label,
+      required,
+      validationErrorMessages,
+      validationErrorsClassName,
+      scrollValidationErrorsIntoView,
+    },
+    ref
+  ) => {
+    const id = React.useId();
+
+    const globals = useArmstrongConfig({
```

Comment:
You'll probably want to run a few more optional local props through globals, stuff like `validationErrorMode` and `validationIcon` etc. Take a look at `input.component.tsx` in module-new

Reply:
It doesn't look like all the globals you've defined local overrides for are being used.

The rule with globals is, if you're control needs to use something that has a global setting, it should define a local prop for it, run it through the useArmstrongConfig hook, and then use the property in the control with `globals.x`, but if it's a global property that you don't need to read then you don't need to do any of this for that prop

Reply:
As the switch isn't using the Input wrapper and (to my knowledge) we're not using an error icon in this component, do we need all those props?